ush/python/pygfs/task/atmens_analysis.py

```diff
@@ -0,0 +1,332 @@
+                'comin_ges_atm': self.config.COMIN_GES,
+                'comin_ges_atmens': self.config.COMIN_GES_ENS,
+                'OPREFIX': f"{self.config.EUPD_CYC}.t{self.runtime_config.cyc:02d}z.",  # TODO: CDUMP is being replaced by RUN
+                'APREFIX': f"{self.runtime_config.CDUMP}.t{self.runtime_config.cyc:02d}z.",  # TODO: CDUMP is being replaced by RUN
+                'GPREFIX': f"gdas.t{self.runtime_config.previous_cycle.hour:02d}z.",
+                'fv3jedi_yaml': _fv3jedi_yaml,
+            }
+        )
+
+        # task_config is everything that this task should need
+        self.task_config = AttrDict(**self.config, **self.runtime_config, **local_dict)
+
+    @logit(logger)
+    def initialize(self: Analysis) -> None:
+        """Initialize a global atmens analysis
+
+        This method will initialize a global atmens analysis using JEDI.
+        This includes:
+        - staging CRTM fix files
+        - staging FV3-JEDI fix files
+        - staging model backgrounds
+        - generating a YAML file for the JEDI executable
+        - linking the JEDI executable (TODO make it copyable, requires JEDI fix)
+        - creating output directories
+
+        Parameters
+        ----------
+        Analysis: parent class for GDAS task
+
+        Returns
+        ----------
+        None
+        """
+        super().initialize()
+
+        # stage CRTM fix files
+        crtm_fix_list_path = os.path.join(self.task_config.HOMEgfs, 'parm', 'parm_gdas', 'atm_crtm_coeff.yaml')
+        logger.debug(f"Staging CRTM fix files from {crtm_fix_list_path}")
+        crtm_fix_list = parse_yamltmpl(crtm_fix_list_path, self.task_config)
+        FileHandler(crtm_fix_list).sync()
+
+        # stage fix files
+        jedi_fix_list_path = os.path.join(self.task_config.HOMEgfs, 'parm', 'parm_gdas', 'atm_jedi_fix.yaml')
+        logger.debug(f"Staging JEDI fix files from {jedi_fix_list_path}")
+        jedi_fix_list = parse_yamltmpl(jedi_fix_list_path, self.task_config)
+        FileHandler(jedi_fix_list).sync()
+
+        # stage backgrounds
+        FileHandler(self.get_bkg_dict(AttrDict(self.task_config))).sync()
+
+        # generate ensemble da YAML file
+        logger.debug(f"Generate ensemble da YAML file: {self.task_config.fv3jedi_yaml}")
+        ensda_yaml = parse_j2yaml(self.task_config.ATMENSYAML, self.task_config)
+        save_as_yaml(ensda_yaml, self.task_config.fv3jedi_yaml)
+        logger.info(f"Wrote ensemble da YAML to: {self.task_config.fv3jedi_yaml}")
+
+        # link executable to DATA/ directory
+        exe_src = self.task_config.JEDIENSEXE
+        logger.debug(f"Link executable {exe_src} to DATA/")  # TODO: linking is not permitted per EE2.  Needs work in JEDI to be able to copy the exec.
+        exe_dest = os.path.join(self.task_config.DATA, os.path.basename(exe_src))
+        if os.path.exists(exe_dest):
+            rm_p(exe_dest)
+        os.symlink(exe_src, exe_dest)
```

Comment:
I think we can do this better by defining `JEDIEXE` in `config.atmanalrun` and `config.atmensanalrun` etc, and lift this to `analysis.py` as a method to `link_jediexe` or something.

Reply:
Suggested change made in working copies of `analysis.py`, `atmens_analysis.py`, and `config.atmensanl`.   Made the following modifications
- Use `JEDIEXE` to define executable in `config.atmensanl`.   Same can be done in `config.atmanl` and `config.aeroanl`.    
- Remove executable link from `atmens_analysis.py` and `atm_analysis.py`.   
- Add the following method to `analysis.py`
```
+    @logit(logger)
+    def link_jediexe(self: Task) -> None:
+        """Compile a dictionary of background error files to copy
+
+        This method links a JEDI executable to the run directory
+
+        Parameters
+        ----------
+        Task: GDAS task
+
+        Returns
+        ----------
+        None
+        """
+        exe_src = self.task_config.JEDIEXE
+
+        # TODO: linking is not permitted per EE2.  Needs work in JEDI to be able to copy the exec.
+        logger.debug(f"Link executable {exe_src} to DATA/")
+        exe_dest = os.path.join(self.task_config.DATA, os.path.basename(exe_src))
+        if os.path.exists(exe_dest):
+            rm_p(exe_dest)
+        os.symlink(exe_src, exe_dest)
+
+        return
```
This method is executed from `initialize` via
```
+        # link jedi executable to run directory
+        self.link_jediexe()
```

Tested both UFS-DA ATM and ENS in prufsatm.  The modified scripts and configs work.  I did not test UFS-DA Aerosol.   

If the above is acceptable, I'll commit the changes to `RussTreadon-NOAA:feature/refactor_jediens`

Reply:
Given that this change can work for UFS-DA Aerosol, ATM, and ENS apps, I committed the required changes to `RussTreadon-NOAA:feature/refactor_jediens` at [4b61129](https://github.com/RussTreadon-NOAA/global-workflow/commit/4b61129abbe17220c51ef028f66d945b1ad40758).